velox/expression/tests/ArgumentTypeFuzzer.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Random.h>
+
+#include "velox/expression/FunctionSignature.h"
+#include "velox/expression/SignatureBinder.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox::test {
+
+// Bind a function signature with a concrete return type. bindings() return a
+// map from each type variable in the signature to the corresponding concrete
+// type if determined. If a type variable is not bounded by the return type, it
+// maps to nullptr.
+class ReverseSignatureBinder {
+ public:
+  ReverseSignatureBinder(const exec::FunctionSignature& signature)
+      : signature_{signature},
+        argumentTypes_{},
+        binder_{signature_, argumentTypes_} {}
+
+  bool tryBindReturnType(const TypePtr& returnType) {
+    std::vector<TypePtr> argumentTypes;
+    if (!binder_.tryBind(signature_.returnType(), returnType)) {
+      return false;
+    }
+    return true;
+  }
+
+  // Return the determined bindings. This function should be called after
+  // tryBindReturnType.
```

Comment:
nit: Perhaps, clarify that "...after tryBind() and only if tryBind() returned 'true'".